src/ansys/aedt/core/visualization/advanced/rcs_visualization.py

```diff
@@ -0,0 +1,1991 @@
+        """Active incident wave Theta."""
+        if self.__incident_wave_theta is None:
+            self.__incident_wave_theta = self.available_incident_wave_theta[0]
+        return self.__incident_wave_theta
+
+    @incident_wave_theta.setter
+    def incident_wave_theta(self, val):
+        """Active incident wave Theta."""
+        if val in self.available_incident_wave_theta:
+            self.__incident_wave_theta = val
+        else:
+            self.__logger.error("Value not available.")
+
+    @property
+    def available_incident_wave_phi(self):
+        """Available incident wave Phi."""
+        if "IWavePhi" in self.raw_data.index.names:
+            self.__available_incident_wave_phi = np.unique(np.array(self.raw_data.index.get_level_values("IWavePhi")))
+        return self.__available_incident_wave_phi
+
+    @property
+    def incident_wave_phi(self):
+        """Active incident wave Phi."""
+        if self.__incident_wave_phi is None:
+            self.__incident_wave_phi = self.available_incident_wave_phi[0]
+        return self.__incident_wave_phi
+
+    @incident_wave_phi.setter
+    def incident_wave_phi(self, val):
+        """Active incident wave Phi."""
+        if val in self.available_incident_wave_phi:
+            self.__incident_wave_phi = val
+        else:
+            self.__logger.error("Value not available.")
+
+    @property
+    def frequency(self):
+        """Active frequency."""
+        return self.__frequency
+
+    @frequency.setter
+    def frequency(self, val):
+        if isinstance(val, str):
+            frequency, units = decompose_variable_value(val)
+            unit_converter(frequency, "Freq", units, self.frequency_units)
+            val = frequency
+        if val in self.frequencies:
+            self.__frequency = val
+            self.__freq_index = self.frequencies.index(val)
+        else:  # pragma: no cover
+            self.__logger.error("Frequency not available.")
+
+    @property
+    def data_conversion_function(self):
+        """RCS data conversion function.
+        The available functions are:
+
+        - `"dB10"`: Converts the data to decibels using base 10 logarithm.
+        - `"dB20"`: Converts the data to decibels using base 20 logarithm.
+        - `"abs"`: Computes the absolute value of the data.
+        - `"real"`: Computes the real part of the data.
+        - `"imag"`: Computes the imaginary part of the data.
+        - `"norm"`: Normalizes the data to have values between 0 and 1.
+        - `"ang"`: Computes the phase angle of the data in radians.
+        - `"ang_deg"`: Computes the phase angle of the data in degrees.
+        """
+        return self.__data_conversion_function
+
+    @data_conversion_function.setter
+    def data_conversion_function(self, val):
+        available_functions = ["dB10", "dB20", "abs", "real", "imag", "norm", "ang", "ang_deg", None]
+        if val in available_functions:
+            self.__data_conversion_function = val
```

Comment:
Can you add a log warning when `val` isn't contained in the available functions ?